IpmiFeaturePkg/Library/IpmiBaseLibMm/IpmiBaseLibMm.inf

```diff
@@ -0,0 +1,33 @@
+## @file
+#
+# @copyright
+# Copyright 2010 - 2021 Intel Corporation. <BR>
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+[Defines]
+  INF_VERSION                    = 1.26
+  PI_SPECIFICATION_VERSION       = 0x00010032
+  BASE_NAME                      = IpmiBaseLibMm
+  FILE_GUID                      = 06C28B6D-F15F-42C3-8316-01BD3E93D2DE
+  MODULE_TYPE                    = MM_STANDALONE
+  VERSION_STRING                 = 1.0
+  LIBRARY_CLASS                  = IpmiBaseLib|MM_STANDALONE MM_CORE_STANDALONE
+
+[sources]
+  IpmiBaseLibMm.c
+
+[Packages]
+  MdePkg/MdePkg.dec
+  IpmiFeaturePkg/IpmiFeaturePkg.dec
+
+[LibraryClasses]
+  DebugLib
+  MmServicesTableLib
+
+[Protocols]
+  gSmmIpmiTransportProtocolGuid
+
+[Depex]
+  TRUE

```

Comment:
Is this right? I thought you need gSmmIpmiTransportProtocolGuid to proceed? Or this is a soft requirement and not sending the command is fine?

Reply:
Actually, this lib is consumed by GenericIpmi drivers, soHi @kuqin12, Actually, this lib is consumed by GenericIpmi drivers that produce the protocol, so if we add the dependency to this library, it will cause the driver never be dispatched. if we add the dependency to this library, it will cause

Reply:
Hmm, then why would the driver not directly call the backend instead of going through the extra layers?

Reply:
I think the IpmiBaseLib is not just used by the GenericIpmi drivers, I can see some other drivers are also consume it, and it is the same implementation, that's the current situation, so we cannot simplify clean up the IpmiBaseLib and put it as internal interface, otherwise we will need to maintain the same implementation in two places.